shim/src/evaluators.js

```diff
@@ -81,7 +81,10 @@ export function getFunctionEvaluator(realmRec) {
 
   const evaluator = function Function(...params) {
     const functionBody = `${params.pop()}` || '';
-    let functionParams = params.join(',');
+    let functionParams = `${params.join(',')}`;
+    // todo: can the Function constructor syntax accept complex parameters
```

Comment:
I don’t think it can since every function evaluates in sloppy mode, and non-simple parameter  lists are only allowed in strict mode?

Reply:
I just did a quick test, it appears that default/rest/destructuring parameters all work just fine. I saw the prohibition on parentheses, but those features all different punctuation. I'll update the test and the comment.

Reply:
Ah, the issue is when the function body contains a strict directive and the parameter list is non-simple.